packages/builder/src/stores/builder/automations.js

```diff
@@ -519,6 +524,10 @@ const automationActions = store => ({
         runtimeName = `loop.${name}`
       } else if (idx === 0) {
         runtimeName = `trigger.${name}`
+      } else if (
+        currentBlock?.stepId === AutomationActionStepId.EXECUTE_SCRIPT
+      ) {
+        runtimeName = `steps["${pathSteps[idx]?.id}"].${name}`
```

Comment:
The presence of the `?` indicates that this might end up being `steps["undefeind"].name`. Would that be a bug? If yes, we likely need checks around this so we can remove the question mark.

Same with the pre-existing line below.

Reply:
think it's more just being defensive here, we just don't want it to error out if somehow `pathSteps[idx] `is undefined. Having a check beforehand is going to result in the same unusable binding regardless.

Reply:
How come we don't want it to error out? Isn't an error explaining the problem preferable to silently generating incorrect code?

Reply:
Because binding generation can continue on for other steps etc. This binding gets inserted onto the user's input. They'll see immediately that it's undefined. I added in some code to surface if this specific binding index was undefined, what do you think?